src/arena/arena_loader.py

```diff
@@ -50,6 +54,37 @@ def load_arena_by_name(name: str) -> Arena:
             arena._spawn_points.append(spawn_point)
             if initial_spawn_for_player is not None:
                 arena._initial_spawn_points[initial_spawn_for_player] = spawn_point
+        elif object.class_ == "PatrolWaypoint":
+            transform, size = get_tile_object_transform_and_size(object)
+            patrol_waypoints[object.id] = PatrolWaypoint(transform[:2])
+
+    # Discover patrol loops
+    patrol_loops: list[list[PatrolWaypoint]] = []
+    while len(patrol_waypoints):
+        patrol_loop: list[PatrolWaypoint] = []
+        patrol_loops.append(patrol_loop)
+        object_id, waypoint = patrol_waypoints.popitem()
+        first_object_id = object_id
+        object = objects_by_id[object_id]
+        patrol_loop.append(waypoint)
+        while True:
+            object_id = int(object.properties["next"])
+            object = objects_by_id[object_id]
+            # If we completed the loop, stop here
+            if object_id == first_object_id:
+                break
+            waypoint = patrol_waypoints.pop(object_id)
+            patrol_loop.append(waypoint)
+    if len(patrol_loops) != 1:
+        raise Exception(
+            "Arena must have exactly one patrol loop.  This restriction will likely be removed in the future."
+        )
+    arena._patrol_loop = Path(
+        [
+            (index * 1.0, point.location[0], point.location[1])
```

Comment:
Non Blocking: As we discussed, it would be more intuitive to give the car a speed that base it on a time between way points. That doesn't need to happen in this PR though.